source/common/upstream/conn_pool_map_impl.h

```diff
@@ -26,13 +29,12 @@ ConnPoolMap<KEY_TYPE, POOL_TYPE>::getPool(KEY_TYPE key, const PoolFactory& facto
   }
 
   // We need a new pool. Check if we have room.
-  if (max_size_.has_value() && size() >= max_size_.value()) {
+  if (!host_->cluster().resourceManager(priority_).connectionPools().canCreate()) {
     // We're full. Try to free up a pool. If we can't, bail out.
     if (!freeOnePool()) {
       return absl::nullopt;
     }
 
-    ASSERT(size() < max_size_.value(), "Freeing a pool should reduce the size to below the max.");
```

Comment:
Can we leave this assert and pull the max size from the resource manager?

Reply:
Yeah. Not sure why I decided this wasn't necessary anymore.